addons/web/models/models.py

```diff
@@ -183,6 +183,8 @@ def cleanup(vals: dict) -> dict:
                     if field.type == 'reference':
                         co_record = record[field_name]
                     else:  # field.type == 'many2one_reference'
+                        if not record[field.model_field]:
+                            continue
```

Comment:
```suggestion
                        if not record[field.model_field]:
                            record_values[field_name] = False
                            continue
```
And you need to move `record_values`. (see my other comment)